cmd/releasego/update-azure-linux.go

```diff
@@ -61,118 +80,190 @@ func updateAzureLinux(p subcmd.ParseFunc) error {
 		return err
 	}
 
-	assets, err := loadBuildAssets(buildAssetJSON)
-	if err != nil {
-		return err
-	}
+	// Set custom user agent to help GitHub identify the bot if necessary.
+	client.UserAgent = "microsoft/go-infra update-azure-linux"
 
-	if updateBranch == "nil" || updateBranch == "" {
-		updateBranch = generateUpdateBranchNameFromAssets(assets)
-	}
-
-	golangSpecFileBytes, err := downloadFileFromRepo(ctx, client, owner, repo, baseBranch, golangSpecFilepath)
+	// Check that the PAT has the necessary scopes.
+	patScopes, err := githubutil.PATScopes(ctx, client)
 	if err != nil {
 		return err
 	}
+	if !slices.Contains(patScopes, "repo") {
+		return fmt.Errorf("the PAT must have 'repo' scope, but not found in list: %v", patScopes)
+	}
 
-	golangSpecFileContent := string(golangSpecFileBytes)
+	if upstream != owner {
+		// Submitting PR via fork. Try to make sure it'll work.
+		if !slices.Contains(patScopes, "workflow") {
+			return fmt.Errorf("the PAT must have 'workflow' scope, but not found in list: %v", patScopes)
+		}
 
-	prevGoArchiveName, err := extractGoArchiveNameFromSpecFile(golangSpecFileContent)
-	if err != nil {
-		return err
+		// Check if fork already exists, or create it.
+		forkRepo, err := githubutil.FetchRepositoryOrNil(ctx, client, owner, repo)
+		if err != nil {
+			return err
+		}
+		if forkRepo == nil {
+			forkRepo, err = githubutil.FullyCreateFork(ctx, client, upstream, repo)
+			if err != nil {
+				return err
+			}
+		}
+		log.Printf("Submitting PR via owner's fork: %s\n", forkRepo.GetHTMLURL())
 	}
 
-	golangSpecFileContent, err = updateSpecFile(assets, start, golangSpecFileContent)
+	assets, err := loadBuildAssets(buildAssetJSON)
 	if err != nil {
 		return err
 	}
 
-	// Validation (as described in previous response)
-	if assets.GoSrcURL == "" || assets.GoSrcSHA256 == "" {
-		return fmt.Errorf("invalid or missing GoSrcURL or GoSrcSHA256 in assets.json")
+	if updateBranch == "nil" || updateBranch == "" {
+		updateBranch = generateUpdateBranchNameFromAssets(assets)
 	}
 
-	golangSignaturesFileBytes, err := downloadFileFromRepo(ctx, client, owner, repo, baseBranch, golangSignaturesFilepath)
-	if err != nil {
-		return err
-	}
+	// If anything fails here, retry from the beginning to use a fresh base commit.
+	// Some individual steps also have their own retries; this is fine.
+	if err := githubutil.Retry(func() error {
+		// Find details about the state of the upstream base branch. This pins the commit that we're
+		// working on, making sure our PR won't revert an unfortunately-timed upstream merge.
+		upstreamRef, _, err := client.Git.GetRef(ctx, upstream, repo, baseBranch)
+		if err != nil {
+			return fmt.Errorf("failed to get ref %v: %v", baseBranch, err)
```

Comment:
`%v` rather than `%w` is actually bad here (and a few other places) because `Retry` needs to handle `*github.RateLimitError`.